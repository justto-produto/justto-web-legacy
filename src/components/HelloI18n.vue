<template>
  <p>{{ $t('hello') }}</p>
</template>

<script>
export default {
  name: 'HelloI18n'
}
</script>

<style media="screen">

</style>

<i18n>
{
  "en": {
    "hello": "Hello i18n in SFC!"
  },
  "pt-BR": {
    "hello": "Ol√° i18n in SFC!"
  }
}
</i18n>
