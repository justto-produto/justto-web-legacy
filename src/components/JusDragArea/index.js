import JusDragArea from './JusDragArea.vue'

export { JusDragArea }
