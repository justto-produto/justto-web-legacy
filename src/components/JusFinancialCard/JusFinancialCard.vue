<template>
  <el-card
    shadow="hover"
    class="jus-financial-card"
  >
    <div class="jus-financial-card__actions">
      <el-tooltip
        v-for="(action, index) in actions"
        :content="action.label"
        :key="index"
        @click.native="action.handler"
      >
        <jus-icon
          :icon="action.icon"
          class="jus-financial-card__icon"
        />
      </el-tooltip>
    </div>

    <span class="jus-financial-card__title">
      {{ title }}
    </span>

    <span class="jus-financial-card__cash">
      R$ {{ value }}
    </span>
  </el-card>
</template>

<script>
export default {
  name: 'JusFinancialCard',
  props: {
    actions: {
      type: Array,
      default: null,
    },
    title: {
      type: String,
      default: '',
    },
    value: {
      type: String,
      default: '00,00',
    },
  },
}
</script>

<style lang="scss" scoped>
@import '@/styles/colors.scss';

.jus-financial-card {
  color: $--color-text-primary;
  text-align: center;

  &:hover {
    .jus-financial-card__actions {
      align-items: center;
      display: grid;
      gap: 8px;
      grid-template-columns: repeat(auto-fit, 16px);
      justify-content: end;
      padding: 0 8px;
      position: absolute;
      right: 0;
      top: 8px;
      width: 100%;

      .jus-financial-card__icon {
        cursor: pointer;
        width: 100%;
      }
    }
  }

  .jus-financial-card__actions {
    display: none;
  }

  .jus-financial-card__title {
    font-size: 16px;
    font-weight: 700;
  }

  .jus-financial-card__cash {
    color: $--color-primary-light-3;
    font-size: 32px;
    font-weight: 700;
  }
}
</style>

<style lang="scss">
.jus-financial-card {
  & > .el-card__body {
    display: grid;
    gap: 8px;
    grid-template-rows: repeat(2, auto);
    padding: 24px;
    position: relative;
  }
}
</style>
