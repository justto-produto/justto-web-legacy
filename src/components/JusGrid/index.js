import JusGrid from './JusGrid'

export { JusGrid }
