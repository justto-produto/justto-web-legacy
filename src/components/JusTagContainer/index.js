import JusTagContainer from './JusTagContainer'

export { JusTagContainer }
