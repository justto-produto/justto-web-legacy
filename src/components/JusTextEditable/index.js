import JusTextEditable from './JusTextEditable'

export { JusTextEditable }
