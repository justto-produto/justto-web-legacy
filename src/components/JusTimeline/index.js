import JusTimeline from './JusTimeline'

export { JusTimeline }
