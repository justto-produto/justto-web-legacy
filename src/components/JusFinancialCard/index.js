import JusFinancialCard from './JusFinancialCard'

export { JusFinancialCard }
