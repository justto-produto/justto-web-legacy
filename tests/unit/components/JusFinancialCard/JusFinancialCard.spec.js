import { JusFinancialCard } from '@/components/JusFinancialCard'
import { shallowMount } from '@vue/test-utils'

describe('Components > JusFinancialCard', () => {
  const wrapper = shallowMount(JusFinancialCard)

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
