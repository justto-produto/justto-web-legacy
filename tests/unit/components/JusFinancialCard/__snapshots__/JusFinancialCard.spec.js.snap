// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > JusFinancialCard renders correctly 1`] = `
<el-card
  class="jus-financial-card"
  shadow="hover"
>
  <div
    class="jus-financial-card__actions"
  />
   
  <span
    class="jus-financial-card__title"
  >
    
    
  
  </span>
   
  <span
    class="jus-financial-card__cash"
  >
    
    R$ 00,00
  
  </span>
</el-card>
`;
